### PR TITLE
Add anomalies api wrapper

### PR DESCRIPTION
**Pull request by tspenov**
**NOTE**:  Should be merged after fixes in tech-indicators for double encoded json and unix timestamps in milliseconds.

Graphql wrapper around metrics anomalies API.

```graphql
{
  metricAnomaly(
    metric: DAILY_ACTIVE_ADDRESSES,
    slug: "ethereum"
    from: "2019-01-01T00:00:00Z",
    to: "2019-06-01T00:00:00Z",
    interval: "1d"
  ) {
    datetime
   metricValue
  }
}

{
  "data": {
    "metricAnomaly": [
      {
        "datetime": "2019-01-15T00:00:00.000Z",
        "metricValue": 240116
      },
      {
        "datetime": "2019-01-16T00:00:00.000Z",
        "metricValue": 211375
      },
      {
        "datetime": "2019-01-17T00:00:00.000Z",
        "metricValue": 210880
      },
      {
        "datetime": "2019-01-18T00:00:00.000Z",
        "metricValue": 214057
      },
```

### CODE REVIEW

File: lib/sanbase_web/graphql/schema/queries/tech_indicators_queries.ex [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    Field `value` is the value from original metric that is considered abnormanl.
+
+    Arguments description:
+    * metric - name of metric (currently supports DAILY_ACTIVE_ADDRESSES and DEV_ACTIVITY)
+    * slug - project's slug
+    * from - a string representation of datetime value according to the iso8601 standard, e.g. "2018-04-16T10:02:19Z"
+    * to - a string representation of datetime value according to the iso8601 standard, e.g. "2018-04-16T10:02:19Z"
+    * interval - an integer followed by one of: `m`, `h`, `d`, `w`
+    """
+    field :metric_anomalies, list_of(:anomaly_value) do
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/schema/queries/tech_indicators_queries.ex]: We consistently name queries singular (`history_price` but not `history_prices`, `token_velocity` but not `token_velocities`, etc.)

File: lib/sanbase_web/graphql/schema/queries/tech_indicators_queries.ex [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+      arg(:metric, non_null(:anomalies_metrics_enum))
+      arg(:slug, non_null(:string))
+      arg(:from, non_null(:datetime))
+      arg(:to, non_null(:datetime))
+      arg(:interval, :string, default_value: "1d")
+
+      complexity(&Complexity.from_to_interval/3)
+      middleware(TimeframeRestriction)
+      resolve(&TechIndicatorsResolver.metric_anomalies/3)
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/schema/queries/tech_indicators_queries.ex]: cache_resolve

File: lib/sanbase/tech_indicators/metric_anomalies.ex [hunk 3]
```diff
@@ -0,0 +1,94 @@
+defmodule Sanbase.TechIndicators.MetricAnomalies do
```
Comment (IvanIvanoff) [inline lib/sanbase/tech_indicators/metric_anomalies.ex]: The module names are also singular in almost all cases

File: lib/sanbase_web/graphql/schema/types/tech_indicators_types.ex [hunk 4]
```diff
@@ -39,4 +39,14 @@ defmodule SanbaseWeb.Graphql.TechIndicatorsTypes do
...
     field(:datetime, non_null(:datetime))
   end
+
+  enum :anomalies_metrics_enum do
+    value(:daily_active_addresses)
+    value(:dev_activity)
+  end
+
+  object :anomaly_value do
+    field(:value, :float)
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/schema/types/tech_indicators_types.ex]: Here we'll need some description and probably rename to `metricValue`. Just `value` sounds like it's a unit a of measuring the anomaly while it's actually just the value of the metric at this point

File: lib/sanbase_web/graphql/schema/queries/tech_indicators_queries.ex [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+      * interval - an integer followed by one of: `m`, `h`, `d`, `w`
+      * from - a string representation of datetime value according to the iso8601 standard, e.g. "2018-04-16T10:02:19Z"
+      * to - a string representation of datetime value according to the iso8601 standard, e.g. "2018-04-16T10:02:19Z"
+      * socialVolumeType - the source of mention counts, one of the following:
+        1. "PROFESSIONAL_TRADERS_CHAT_OVERVIEW" - shows how many times the given project has been mentioned in the professional traders chat
+        2. "TELEGRAM_CHATS_OVERVIEW" - shows how many times the given project has been mentioned across all telegram chats, except the project's own community chat (if there is one)
+        3. "TELEGRAM_DISCUSSION_OVERVIEW" - the general volume of messages in the project's community chat (if there is one)
+        4. "DISCORD_DISCUSSION_OVERVIEW" - shows how many times the given project has been mentioned in the discord channels
+    """
+    field :social_volume, list_of(:social_volume) do
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/schema/queries/tech_indicators_queries.ex]: let's have this is `social_queries.ex`

File: lib/sanbase_web/graphql/schema/types/tech_indicators_types.ex (hunk 4 above)
Comment (tspenov) [reply lib/sanbase_web/graphql/schema/types/tech_indicators_types.ex]: yep, sounds good. I also was confused that is the anomaly value at first.
About description you mean explaining why the value is considered abnormal ?

File: lib/sanbase_web/graphql/schema/queries/tech_indicators_queries.ex [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+        1. TELEGRAM
+        2. PROFESSIONAL_TRADERS_CHAT
+        3. REDDIT
+        4. DISCORD
+      * searchText - a string containing the key words for which the sources should be searched.
+      * interval - an integer followed by one of: `m`, `h`, `d`, `w`
+      * from - a string representation of datetime value according to the iso8601 standard, e.g. "2018-04-16T10:02:19Z"
+      * to - a string representation of datetime value according to the iso8601 standard, e.g. "2018-04-16T10:02:19Z"
+    """
+    field :topic_search, :topic_search do
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/schema/queries/tech_indicators_queries.ex]: These (and some others of the queries here) belong to `social_data_queries.ex`
Separating by concern and not by source is better. For example when we moved from from timescale to clickhouse for some metrics if we had separate by `source` we would have to move some queries for the 3rd time (schema -> timescaledb -> clickhouse)

File: lib/sanbase_web/graphql/schema/queries/tech_indicators_queries.ex [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+    Field `value` is the value from original metric that is considered abnormanl.
+
+    Arguments description:
+    * metric - name of metric (currently supports DAILY_ACTIVE_ADDRESSES and DEV_ACTIVITY)
+    * slug - project's slug
+    * from - a string representation of datetime value according to the iso8601 standard, e.g. "2018-04-16T10:02:19Z"
+    * to - a string representation of datetime value according to the iso8601 standard, e.g. "2018-04-16T10:02:19Z"
+    * interval - an integer followed by one of: `m`, `h`, `d`, `w`
+    """
+    field :metric_anomaly, list_of(:anomaly_value) do
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/schema/queries/tech_indicators_queries.ex]: This though can stay here in my opinion as there isn't any other similar that can be grouped with this one

File: lib/sanbase_web/graphql/schema/queries/tech_indicators_queries.ex (hunk 5 above)
Comment (tspenov) [reply lib/sanbase_web/graphql/schema/queries/tech_indicators_queries.ex]: I can move some of the queries to social_data_queries. Some of them are not - like price-volume-diff, anomalies

File: lib/sanbase_web/graphql/schema/queries/tech_indicators_queries.ex (hunk 6 above)
Comment (tspenov) [reply lib/sanbase_web/graphql/schema/queries/tech_indicators_queries.ex]: agree that these are for soacial_data_queries

File: lib/sanbase_web/graphql/schema/queries/tech_indicators_queries.ex (hunk 7 above)
Comment (tspenov) [reply lib/sanbase_web/graphql/schema/queries/tech_indicators_queries.ex]: yes this one and price-volume should stay

File: lib/sanbase_web/graphql/schema/queries/tech_indicators_queries.ex (hunk 5 above)
Comment (IvanIvanoff) [reply lib/sanbase_web/graphql/schema/queries/tech_indicators_queries.ex]: Yes, check my comment under the `metric_anomaly` query

**Pull request merged**